### PR TITLE
Improve trend line graph UI/UX

### PR DESCRIPTION
**Pull request by SethCKL**
this PR seeks to address the issues raised in #89 and #87, and all arising issues w.r.t the TimeBasedChart

### CODE REVIEW

**Comment by SethCKL** [comment]
This also requires new endpoints to be supplied from the BE to provide the 3 following filtered views at least 

1. Entire History 
2. Last X number of years?
3. Last 12 Months

Some thought is required on how this is done, because enough data points need to be had in order for the trendline to be useful

we aim for at least 12 data points (one for every month), however this is complicated when we want 1. and 2.

This means the BE needs to handle this complexity, based on how much data is available to generate it in a certain form.

**Comment by SethCKL** [comment]
For example, for the first view (entire history)

Place A has 3 years of data (24 months) Place B has 6 years of data.

Place A's trend line should be generated as 12 quarters? Place B's trend line should be 12 half years?

**Comment by SethCKL** [comment]
@YaleChen299 tagging you to get your opinion on the above

**Comment by zhekaiii** [comment]
Actually should we just do away with the dots
image
image

**Comment by SethCKL** [comment]
> Actually should we just do away with the dots?

Yes i think we should, looks much cleaner!

File: src/components/data/TimeBasedChart.tsx [hunk 1]
```diff
@@ -30,8 +30,8 @@ export const TimeBasedChart: React.FC<Props> = ({ data, title }) => {
         legendProps={{ verticalAlign: "bottom", height: 50 }}
         curveType="monotone"
         strokeWidth={2}
-        dotProps={{ r: 1, strokeWidth: 1, stroke: "#fff" }}
-        activeDotProps={{ r: 2, strokeWidth: 2, stroke: "#fff" }}
+        dotProps={{ r: 0 }}
+        activeDotProps={{ r: 0 }}
```
Comment (zhekaiii) [inline src/components/data/TimeBasedChart.tsx]: I'll keep the `activeDotProps` since it still makes sense to keep these.

**Review by zhekaiii — APPROVED** [review]
LGTM, thanks!

**Pull request merged**